src/components/DataTable/DataTable.tsx

```diff
@@ -32,28 +33,38 @@ import {
   RevokeKeyDialogProps,
   TransactionEditDialogProps,
   DestinationEditDialogProps,
+  PaymailDeleteDialogProps,
 } from '@/components';
+import { AccessKey, Contact, Destination, PaymailAddress, Tx, XPub } from '@bsv/spv-wallet-js-client';
+
+export type RowType = XPub | Contact | AccessKey | Destination | PaymailAddress | Tx;
 
 interface DataTableProps<TData, TValue> {
   columns: ColumnDef<TData, TValue>[];
   data: TData[];
-  EditDialog?: React.ComponentType<ContactEditDialogProps | TransactionEditDialogProps | DestinationEditDialogProps>;
+  ContactEditDialog?: React.ComponentType<ContactEditDialogProps>;
+  TransactionEditDialog?: React.ComponentType<TransactionEditDialogProps>;
+  DestinationEditDialog?: React.ComponentType<DestinationEditDialogProps>;
```

Comment:
Why this change? it looks ugly and cumbersome.

Reply:
Related to `any` issue. We can have one complex component with bunch of `if else` statements that will be hard to read and make changes even harder. Or have a few smaller components with straightforward logic.